docs/guides/benchmarking.md

```diff
@@ -0,0 +1,106 @@
+---
+bodyclass: docs
+layout: docs
+headline: gRPC Performance Benchmarking
+sidenav: doc-side-tutorial-nav.html
+type: markdown
+---
+<p class="lead">gRPC is designed to support high-performance
+open-source RPCs in many languages. This document describes the
+performance benchmarking tools, the scenarios considered by the tests,
+and the testing infrastructure.</p>
+
+<div id="toc"></div>
+
+<a name="Overview"></a>
+## Overview
+
+gRPC is designed for both high-performance and high-productivity
+design of distributed applications. Continuous performance
+benchmarking is a critical part of the gRPC development
+workflow. Multi-language performance tests run hourly, and these numbers
+are reported to a dashboard for visualization.
+
+  * [Multi-language performance dashboard @master](https://performance-dot-grpc-testing.appspot.com/explore?dashboard=5760820306771968)
+  * [C++ detailed performance dashboard @master](https://performance-dot-grpc-testing.appspot.com/explore?dashboard=5685265389584384)
+
+## Performance testing design
+
+Each language implements a performance testing worker that implements
+a gRPC
+[WorkerService](https://github.com/grpc/grpc/blob/master/src/proto/grpc/testing/services.proto). This
+service directs the worker to act as either a client or a server for
+the actual benchmark test, represented as
+[BenchmarkService](https://github.com/grpc/grpc/blob/master/src/proto/grpc/testing/services.proto). That
+service has two methods:
+
+  * UnaryCall - a unary RPC of a simple request that specifies the number of bytes to return in the response
+  * StreamingCall - a streaming RPC that allows repeated ping-pongs of request and response messages akin to the UnaryCall
+
+![gRPC performance testing worker diagram](testing_framework.png)
+
+These workers are controlled by a
+[driver](https://github.com/grpc/grpc/blob/master/test/cpp/qps/qps_json_driver.cc)
+that takes as input a scenario description (in JSON format) and an
+environment variable specifying the host:port of each worker process.
+
+
+<a name="Languages under test"></a>
+## Languages under test
+
+The following languages have continuous performance testing as both
+clients and servers at master:
```

Comment:
Come to think of it, the performance dashboard pertains to master, so I am not sure if the earlier rules apply.